File: java-version/src/main/java/bernard/tatin/tools/ForFiles.java
package bernard.tatin.tools;

import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;

import java.util.Arrays;
import org.apache.commons.lang3.ArrayUtils;
import java.util.stream.Stream;

import org.apache.commons.lang3.ArrayUtils;


public class ForFiles {
    public static String loadTextFile(Path path) {
        try {
            Byte tableau[] = ArrayUtils.toObject(Files.readAllBytes(path));
            tableau = Arrays.stream(tableau).map(b -> b == 0 ? 32 : b).toArray(Byte[]::new);
            return new String(ArrayUtils.toPrimitive(tableau));
        }
        catch (Exception e) {
            return null;
        }
    }

    public static String[] loadLinesFromFiles(Path path, String separators) {
        String fileContent = ForFiles.loadTextFile(path);
        if (fileContent != null) {
            return fileContent.split(separators);
        } else {
            return null;
        }
    }
    private ForFiles() {
    }
}

File: kotlin-version/src/main/kotlin/bernard/tatin/WorkInProgress/tools/ForFiles.kt
package bernard.tatin.Tools.tools

import java.io.File
import java.io.InputStream
import java.nio.file.Files
import java.nio.file.Path

// a lot of things to learn here about collections,
// types and map
object ForFiles {
    fun loadTextFile(path: String): String? {
        try {
            val file = File(path)
            val fileBytes : ByteArray = file.inputStream().readBytes()
            val byteCount = fileBytes.size

            if (byteCount > 0) {
                // '\0' is a word separator
                return CharArray(
                        byteCount,
                        fun(index : Int) : Char =
                                if (fileBytes[index].toInt() == 0)
                                    32.toChar()
                                else
                                    fileBytes[index].toChar())
                        .fold(
                                "",
                                fun(acc : String, cc: Char) : String =
                                        "${acc}${cc}"
                        )
            } else {
                println("ERROR: readBytes -> null")
                return null
            }
        } catch (e: Exception) {
            println("ERROR: readBytes -> ${e::toString}")
            return null
        }

    }

    fun loadLinesFromFiles(path: String, separators: String): List<String>? {
        return ForFiles.loadTextFile(path)?.split(separators)
    }
}
